cocos/core/data/class.ts

```diff
@@ -455,10 +455,20 @@ function parseAttributes (constructor: Function, attributes: PropertyStash, clas
         else if (typeof type === 'object') {
             if (Enum.isEnum(type)) {
                 (attrs || initAttrs())[`${propertyNamePrefix}type`] = ENUM_TAG;
-                attrs![`${propertyNamePrefix}enumList`] = Enum.getList(type);
+                Object.defineProperty(attrs!, `${propertyNamePrefix}enumList`, {
+                    enumerable: true,
+                    get () {
+                        return Enum.getList(type);
```

Comment:
Here is the equivalent of caching the `type` variable, is this really good? I prefer not to have this kind of magic. If you want to update, provide the updated API honestly.
This may also break compatibility if someone reuses an obj to generate class ....

Reply:
So instead of caching the `type`, it would be just as effective to just update \_\_bitmask\_\_ instead of assigning it to a new array. This way no extra objects are cached and the data is updated.

Reply:
done modified the code